### PR TITLE
Add toString() to MultiSearchRequest class.

### LINKED ISSUE #12144: [BUG] Missing toString() method in many requests classes
### Describe the bug

Whereas SearchRequest has a toString() method, making debugging much easier, it is not present in all other requests.

For instance it is missing in
- MultiSearchRequest
- CreateIndexRequest
- MultiGetRequest

### Related component

Clients

### To Reproduce

1. Create a MultiSearchRequest object
2. Try to log it
3. See something like org.opensearch.action.search.MultiSearchRequest@3eb87ae4

### Expected behavior

1. Create a MultiSearchRequest object
2. Try to log it
3. See something like `{requests=[SearchRequest{...},...]}`

### Additional Details

**Plugins**
N/A

**Screenshots**
N/A

**Host/Environment (please complete the following information):**
 - any Java client

**Additional context**
I will try to provide a fix in PR.

### PR DESCRIPTION
**Pull request by guillaume-alvarez**
fix #12144

### Description
Make debugging easier with toString() method in MultiSearchRequest class.

### Related Issues
Resolves #12144

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by deshsidd — APPROVED** [review]
LGTM. Although the tests seem to be failing. Please fix or re-trigger the build

**Comment by guillaume-alvarez** [comment]
> LGTM. Although the tests seem to be failing. Please fix or re-trigger the build

Thanks!

The tests failed because of timeouts so I think it is not related to the changes. Could you please tell me what is the best way of re-triggerring the build? Pushing the branch again?

**Review by kkmr — APPROVED** [review]

**Comment by dblock** [comment]
> > LGTM. Although the tests seem to be failing. Please fix or re-trigger the build
> 
> Thanks!
> 
> The tests failed because of timeouts so I think it is not related to the changes. Could you please tell me what is the best way of re-triggerring the build? Pushing the branch again?

Yes, a force push with re-trigger it. Do examine test failures and link to flaky test bugs. Yes, it's a major pain.

**Comment by dblock** [comment]
Is there a way to add a test that would fail for the missing `toString()` implementations across the range of similar classes? Or change the interface requiring it?

**Comment by guillaume-alvarez** [comment]
> Is there a way to add a test that would fail for the missing `toString()` implementations across the range of similar classes? Or change the interface requiring it?

I think the easiest way would be some sonarqube rule or similar tool.

**Comment by guillaume-alvarez** [comment]
> > > LGTM. Although the tests seem to be failing. Please fix or re-trigger the build
> > 
> > 
> > Thanks!
> > The tests failed because of timeouts so I think it is not related to the changes. Could you please tell me what is the best way of re-triggerring the build? Pushing the branch again?
> 
> Yes, a force push with re-trigger it. Do examine test failures and link to flaky test bugs. Yes, it's a major pain.

I checked and the failing tests are different every time:

### Build [33273](https://build.ci.opensearch.org/job/gradle-check/33362/testReport/):
```
 org.opensearch.cluster.MinimumClusterManagerNodesIT.testThreeNodesNoClusterManagerBlock
 org.opensearch.cluster.coordination.AwarenessAttributeDecommissionIT.testConcurrentDecommissionAction
 org.opensearch.cluster.coordination.AwarenessAttributeDecommissionIT.testConcurrentDecommissionAction
 org.opensearch.cluster.coordination.AwarenessAttributeDecommissionIT.testConcurrentDecommissionAction
 org.opensearch.cluster.coordination.AwarenessAttributeDecommissionIT.testConcurrentDecommissionAction
 org.opensearch.search.fetch.subphase.highlight.HighlighterSearchIT.testHighlightNoMatchSize {p0={"search.concurrent_segment_search.enabled":"true"}}
 org.opensearch.search.fetch.subphase.highlight.HighlighterSearchIT.classMethod
```

### Build [33362](https://build.ci.opensearch.org/job/gradle-check/33362/testReport/):
```
 org.opensearch.repositories.azure.AzureBlobStoreRepositoryTests.testMultipleSnapshotAndRollback
 org.opensearch.repositories.azure.AzureBlobStoreRepositoryTests.classMethod
 org.opensearch.repositories.azure.AzureBlobStoreRepositoryTests.testMultipleSnapshotAndRollback
 org.opensearch.repositories.azure.AzureBlobStoreRepositoryTests.classMethod
 org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=cat.shards/10_basic/Help}
 org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/11_indices_metrics/Metric - _all include_segment_file_sizes}
 org.opensearch.cluster.MinimumClusterManagerNodesIT.testThreeNodesNoClusterManagerBlock
 org.opensearch.remotestore.multipart.RemoteStoreMultipartIT.testRequestDurabilityWhenRestrictSettingImplicitFalse
 org.opensearch.search.SearchWeightedRoutingIT.testMultiGetWithNetworkDisruption_FailOpenEnabled
 org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/20_response_filtering/Nodes Stats with response filtering}
 org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/11_indices_metrics/Metric - indices _all}
 org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=cat.nodes/10_basic/Test cat nodes output}
 org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=cat.nodes/10_basic/Additional disk information}
 org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/40_store_stats/Store stats}
 org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/20_response_filtering/Nodes Stats filtered using both includes and excludes filters}
 org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/11_indices_metrics/Metric - multi}
 org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/10_basic/Nodes stats level}
 org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/11_indices_metrics/Metric - _all}
```

The AzureBlobStoreRepositoryTests failed because of test suite timeout:
`java.lang.Exception: Test abandoned because suite timeout was reached.`
Not sure it deserves an issue?

There is already 
- #7755 for MixedClusterClientYamlTestSuiteIT
- #10006 for MinimumClusterManagerNodesIT
- #11997 for RemoteStoreMultipartIT
- #10755  for SearchWeightedRoutingIT

**Comment by dblock** [comment]
@guillaume-alvarez thanks for digging into flakes, you can force push a change to re-kick gradle check

**Comment by guillaume-alvarez** [comment]
[org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/50_indexing_pressure/Indexing pressure memory limit}](https://build.ci.opensearch.org/job/gradle-check/33381/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_50_indexing_pressure_Indexing_pressure_memory_limit_/)
    [org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/11_indices_metrics/Metric - indices _all}](https://build.ci.opensearch.org/job/gradle-check/33381/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_11_indices_metrics_Metric___indices__all_/)
    [org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/60_transport_stats/Transport stats}](https://build.ci.opensearch.org/job/gradle-check/33381/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_60_transport_stats_Transport_stats_/)
    [org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/20_response_filtering/Nodes Stats filtered using both includes and excludes filters}](https://build.ci.opensearch.org/job/gradle-check/33381/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_20_response_filtering_Nodes_Stats_filtered_using_both_includes_and_excludes_filters_/)
    [org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=cat.nodes/10_basic/Test cat nodes output}](https://build.ci.opensearch.org/job/gradle-check/33381/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_cat_nodes_10_basic_Test_cat_nodes_output_/)
    [org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/11_indices_metrics/Metric - _all}](https://build.ci.opensearch.org/job/gradle-check/33381/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_11_indices_metrics_Metric____all_/)
    [org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/40_store_stats/Store stats}](https://build.ci.opensearch.org/job/gradle-check/33381/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_40_store_stats_Store_stats_/)
    [org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/20_response_filtering/Nodes Stats with response filtering}](https://build.ci.opensearch.org/job/gradle-check/33381/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_20_response_filtering_Nodes_Stats_with_response_filtering_/)
    [org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=cat.fielddata/10_basic/Test cat fielddata output}](https://build.ci.opensearch.org/job/gradle-check/33381/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_cat_fielddata_10_basic_Test_cat_fielddata_output_/)
    [org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/11_indices_metrics/Metric - indexing doc_status}](https://build.ci.opensearch.org/job/gradle-check/33381/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_11_indices_metrics_Metric___indexing_doc_status_/)
    [org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/10_basic/Nodes stats level}](https://build.ci.opensearch.org/job/gradle-check/33381/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_10_basic_Nodes_stats_level_/)
    [org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/30_discovery/Discovery stats}](https://build.ci.opensearch.org/job/gradle-check/33381/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_30_discovery_Discovery_stats_/)
    [org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=cat.allocation/10_basic/Node ID}](https://build.ci.opensearch.org/job/gradle-check/33381/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_cat_allocation_10_basic_Node_ID_/)
    [org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/11_indices_metrics/Metric - _all include_segment_file_sizes}](https://build.ci.opensearch.org/job/gradle-check/33381/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_11_indices_metrics_Metric____all_include_segment_file_sizes_/)

They are all covered by #7755

File: CHANGELOG.md [hunk 1]
```diff
@@ -86,6 +86,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Fix typo in API annotation check message ([11836](https://github.com/opensearch-project/OpenSearch/pull/11836))
 - Update supported version for must_exist parameter in update aliases API ([#11872](https://github.com/opensearch-project/OpenSearch/pull/11872))
 - [Bug] Check phase name before SearchRequestOperationsListener onPhaseStart ([#12035](https://github.com/opensearch-project/OpenSearch/pull/12035))
+- As toString methods to MultiSearchRequest, MultiGetRequest and CreateIndexRequest ([#12163](https://github.com/opensearch-project/OpenSearch/pull/12163))
```
Comment (VachaShah) [inline CHANGELOG.md:89]: *Nit*: `Add toString`. Also, this changelog line would go in the Unreleased 2.x section.

Comment (guillaume-alvarez) [reply CHANGELOG.md]: Thanks, I fixed it!

**Review by dblock — APPROVED** [review]

**Review by VachaShah — APPROVED** [review]

**Review by kotwanikunal — APPROVED** [review]

**Comment by dblock** [comment]
@guillaume-alvarez will need a manual backport to get this into a 2.x, please?

**Comment by guillaume-alvarez** [comment]
> @guillaume-alvarez will need a manual backport to get this into a 2.x, please?

Ok, I created https://github.com/opensearch-project/OpenSearch/pull/12314

**Pull request merged**